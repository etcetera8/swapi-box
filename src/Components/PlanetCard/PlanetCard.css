.PlanetCard {
  background: rgba(0, 0, 255, 0.1);
}