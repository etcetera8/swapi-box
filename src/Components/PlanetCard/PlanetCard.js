import React from 'react';
import PropTypes from 'prop-types';

const PlanetCard = ({name, climate, terrain, population, residents, setFavorite, className}) => {
  let resString = ''
  if (residents.length === 0) {
    resString = 'N/A'
  } else {
    resString = residents.join(", ")
  }

  return (
    <article className='Card'> 
      <header>
        <h1 className="name">{name}</h1>
        <button onClick={() => setFavorite(name, 'planets')} className={`favorite ${className}`}>x</button>
      </header>
      <ul className="categories">
        <li>Climate {climate}</li>
        <li>Terrain: {terrain}</li>
        <li>Population: {population}</li>
        <li>Residents: {resString}</li>
      </ul> 
    </article>
  )
}

PlanetCard.propTypes = {
  name: PropTypes.string.isRequired,
  climate: PropTypes.string.isRequired,
  terrain: PropTypes.string.isRequired,
  population: PropTypes.string.isRequired,
  residents: PropTypes.array.isRequired,
  setFavorite: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired
}

export default PlanetCard;