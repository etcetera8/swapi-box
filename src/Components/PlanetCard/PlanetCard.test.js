import React from 'react';
import ReactDOM from 'react-dom';
import PlanetCard from './PlanetCard';
import { shallow } from 'enzyme';

describe('PlanetCard', () => {
  let wrapper;
  const mockFunction = jest.fn();
  const mockPlanet = {name:'naboo', climate:'desert', terrain:'dunes', population: '20', residents: ['liea', 'darth', 'luke']}
  beforeEach( () => {
    wrapper = shallow(<PlanetCard
                        className={'activepeople'}
                        setFavorite={mockFunction}
                        planet={mockPlanet}
                        key={1}
                      />)
  })

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  })
})