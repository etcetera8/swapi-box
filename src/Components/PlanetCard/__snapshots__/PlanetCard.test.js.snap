// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetCard should match the snapshot test 1`] = `
<article
  className="Card PlanetCard"
>
  <header>
    <h1
      className="name"
    >
      naboo
    </h1>
    <button
      className="favorite activepeople"
      onClick={[Function]}
    >
      â˜…
    </button>
  </header>
  <ul
    className="categories"
  >
    <li>
      <span
        className="key"
      >
        Climate:
      </span>
      desert
    </li>
    <li>
      <span
        className="key"
      >
        Terrain:
      </span>
        
      dunes
    </li>
    <li>
      <span
        className="key"
      >
        Population:
      </span>
       
      20
    </li>
    <li>
      <span
        className="key"
      >
        Residents:
      </span>
       
      liea, darth, luke
    </li>
  </ul>
</article>
`;
